test/testdialog.c

```diff
@@ -0,0 +1,149 @@
+        }
+
+        SDL_Log("Filter used: '%s'\n", filter_name);
+
+        while (*files) {
+            SDL_Log("'%s'\n", *files);
+            files++;
+        }
+    } else {
+        SDL_Log("Error: %s\n", SDL_GetError());
+    }
+}
+
+int main(int argc, char *argv[]) {
+    SDL_Window *w;
+    SDL_Renderer *r;
+    SDLTest_CommonState *state;
+    const SDL_FRect open_file_rect = { 50, 50, 220, 140 };
+    const SDL_FRect save_file_rect = { 50, 290, 220, 140 };
+    const SDL_FRect open_folder_rect = { 370, 50, 220, 140 };
+    int i;
+
+    /* Initialize test framework */
+    state = SDLTest_CommonCreateState(argv, 0);
+    if (state == NULL) {
+        return 1;
+    }
+
+    /* Enable standard application logging */
+    SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
+
+    /* Parse commandline */
+    for (i = 1; i < argc;) {
+        int consumed;
+
+        consumed = SDLTest_CommonArg(state, i);
+        if (!consumed) {
+        }
+        if (consumed <= 0) {
+            static const char *options[] = { NULL };
+            SDLTest_CommonLogUsage(state, argv[0], options);
+            return 1;
+        }
+
+        i += consumed;
+    }
+
+    if (SDL_Init(SDL_INIT_VIDEO) < 0) {
+        SDL_Log("SDL_Init failed (%s)", SDL_GetError());
+        return 1;
+    }
+    if (SDL_CreateWindowAndRenderer(640, 480, 0, &w, &r) < 0) {
+        SDL_Log("Failed to create window and/or renderer: %s\n", SDL_GetError());
+        return 1;
+    }
+
+    while (1) {
+        int quit = 0;
+        SDL_Event e;
+        while (SDL_PollEvent(&e)) {
+            if (e.type == SDL_EVENT_QUIT) {
+                quit = 1;
+                break;
+            } else if (e.type == SDL_EVENT_MOUSE_BUTTON_UP) {
+                const SDL_FPoint p = { e.button.x, e.button.y };
+                /*
+                 * Arguments, in order:
+                 * - A function to call when files are chosen (or dialog is canceled, or error happens)
+                 * - A user-managed void pointer to pass to the function when it will be invoked
+                 * - The window to bind the dialog to, or NULL if none
+                 * - A list of filters for the files, see SDL_DialogFileFilter above (not for SDL_ShowOpenFolderDialog)
+                 * - The path where the dialog should start. May be a folder or a file
+                 * - Nonzero if the user is allowed to choose multiple entries (not for SDL_ShowSaveFileDialog)
+                 */
+                if (SDL_PointInRectFloat(&p, &open_file_rect)) {
+                    SDL_ShowOpenFileDialog(callback, NULL, w, filters, "/home/", 1);
+                } else if (SDL_PointInRectFloat(&p, &open_folder_rect)) {
+                    SDL_ShowOpenFolderDialog(callback, NULL, w, "/home/", 1);
+                } else if (SDL_PointInRectFloat(&p, &save_file_rect)) {
+                    SDL_ShowSaveFileDialog(callback, NULL, w, filters, "/home/");
+                }
```

Comment:
Perhaps use ` SDL_GetUserFolder(SDL_FOLDER_HOME)` instead of `/home`?
If you want to get fancy, you can make the path an (optional) (positional) argument.

Reply:
That would help make it cross-platform, I don't think Windows has a `/home` :D